functions/src/firestore-types/users.ts

```diff
@@ -36,6 +36,11 @@ export type UserData = {
 };
 
 export function displayNameValidator(displayName: string): { valid: true } | { valid: false; reason: string } {
+  let newname = displayName;
+  while (newname.charAt(newname.length - 1) === " ") {
+    newname = newname.slice(0, -1);
+  }
+  displayName = newname;
   if (displayName === "") {
```

Comment:
you want to trim the name in the parameter not in this function. Change this back to the original